.github/workflows/e2e-pull-request.yml

```diff
@@ -0,0 +1,57 @@
+name: E2E Tests - Pull Request
+
+on:
+  pull_request:
+    branches:
+      - develop
+      - trunk
+  workflow_dispatch:
+
+env:
+  E2E_GH_TOKEN:          ${{ secrets.E2E_GH_TOKEN }}
+  WCP_DEV_TOOLS_REPO:    ${{ secrets.WCP_DEV_TOOLS_REPO }}
+  WCP_DEV_TOOLS_BRANCH:  'trunk'
+  WCP_SERVER_REPO:       ${{ secrets.WCP_SERVER_REPO }}
+  WC_SUBSCRIPTIONS_REPO: ${{ secrets.WC_SUBSCRIPTIONS_REPO }}
+  E2E_BLOG_ID:           ${{ secrets.E2E_BLOG_ID }}
+  E2E_BLOG_TOKEN:        ${{ secrets.E2E_BLOG_TOKEN }}
+  E2E_USER_TOKEN:        ${{ secrets.E2E_USER_TOKEN }}
+  WC_E2E_SCREENSHOTS:    1
+  E2E_SLACK_CHANNEL:     ${{ secrets.E2E_SLACK_CHANNEL }}
+  E2E_SLACK_TOKEN:       ${{ secrets.E2E_SLACK_TOKEN }}
+  E2E_USE_LOCAL_SERVER:  false
+  E2E_RESULT_FILEPATH:   'tests/e2e/results.json'
+
+
+jobs:
+  wcpay-e2e-tests:
+    runs-on: ubuntu-20.04
+
+    strategy:
+      fail-fast:     false
+      matrix:
+        woocommerce:   [ 'latest' ]
+        wordpress:     [ 'latest' ]
+        test_groups:   [ 'wcpay', 'subscriptions', 'blocks' ]
```

Comment:
Thinking out loud but do we need to run `blocks` related tests on PR?

Reply:
It's more like a compatibility test. Also because we have blocks integration in WCPay